### PR TITLE
add store.dispatch fixes #1644

### LINKED ISSUE #1644: TypeError: _dispatch is not a function
Here's the error:

```
[1] TypeError: _dispatch is not a function
[1]     at Object.dispatch (/Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/redux/lib/applyMiddleware.js:45:18)
[1]     at /Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/react-router-redux/lib/index.js:90:13
[1]     at /Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/history/lib/useQueries.js:109:9
[1]     at /Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/history/lib/useBasename.js:84:9
[1]     at Object.listen (/Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/history/lib/createHistory.js:104:7)
[1]     at Object.listen (/Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/history/lib/useBasename.js:83:22)
[1]     at Object.listen (/Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/history/lib/useQueries.js:108:22)
[1]     at middleware (/Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/react-router-redux/lib/index.js:83:34)
[1]     at /Users/thomas/Desktop/react-redux-universal-hot-example/node_modules/redux/lib/applyMiddleware.js:49:16
[1]     at Array.map (native)
```

Seems that a default value was removed in a commit (https://github.com/reactjs/redux/commit/9496fd793bac83cedb643aba8da2c164aaf52a15) [11 days ago](https://github.com/reactjs/redux/commit/9496fd793bac83cedb643aba8da2c164aaf52a15).

[This line here](https://github.com/reactjs/redux/blob/master/src/applyMiddleware.js#L27) needs access to `dispatch` and it's undefined.

Here's how to recreate the error:

```
git clone https://github.com/erikras/react-redux-universal-hot-example.git
cd react-redux-universal-hot-example
npm install
npm run dev
```

### PR DESCRIPTION
**Pull request by reggi**

### CODE REVIEW

File: src/applyMiddleware.js [hunk 1]
```diff
@@ -18,17 +18,13 @@ import compose from './compose'
  */
 export default function applyMiddleware(...middlewares) {
   return (createStore) => (reducer, initialState, enhancer) => {
-    var store = createStore(reducer, initialState, enhancer)
-    var dispatch
-    var chain = []
```
Comment (reggi) [inline src/applyMiddleware.js]: This chain declaration here doesn't do a thing :/

File: src/applyMiddleware.js [hunk 2]
```diff
@@ -18,17 +18,13 @@ import compose from './compose'
  */
 export default function applyMiddleware(...middlewares) {
   return (createStore) => (reducer, initialState, enhancer) => {
-    var store = createStore(reducer, initialState, enhancer)
-    var dispatch
```
Comment (reggi) [inline src/applyMiddleware.js]: Calling dispatch up here is just confusing.

**Comment by hackerwins** [comment]
@reggi 👍

**Comment by gaearon** [comment]
This doesn’t pass the tests.

**Comment by gaearon** [comment]
I’m closing because of the failing test. #1647 is another fix, and it doesn’t fail the tests.

**Pull request closed without merging**